src/controllers/dynakube/istio/reconciler_test.go

```diff
@@ -72,7 +72,46 @@ func TestIstioClient_BuildDynatraceVirtualService(t *testing.T) {
 }
 
 func TestController_ReconcileIstio(t *testing.T) {
-	server := httptest.NewServer(http.HandlerFunc(reconcileTestHandler))
+	testReconcileIstio(t, true)
+}
+
+func TestController_ReconcileIstioGracefullyFail(t *testing.T) {
+	testReconcileIstio(t, false)
+}
+
+func testReconcileIstio(t *testing.T, enableIstioGVR bool) {
+	server := httptest.NewServer(createReconcileTestHandler(enableIstioGVR))
+	defer server.Close()
+
+	serverUrl, err := url.Parse(server.URL)
+	require.NoError(t, err)
+
+	port, err := strconv.ParseUint(serverUrl.Port(), 10, 32)
+	require.NoError(t, err)
+
+	virtualService := buildVirtualService(buildObjectMeta(testVirtualServiceName, DefaultTestNamespace), "localhost", serverUrl.Scheme, uint32(port))
+	instance := &dynatracev1beta1.DynaKube{
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			APIURL: serverUrl.String(),
+		},
+	}
+	reconciler := Reconciler{
+		istioClient: fakeistio.NewSimpleClientset(virtualService),
+		scheme:      scheme.Scheme,
+		config: &rest.Config{
+			Host:    server.URL,
+			APIPath: testApiPath,
+		},
+	}
+
+	updated, err := reconciler.Reconcile(instance, []dtclient.CommunicationHost{})
+
+	assert.NoError(t, err)
+	assert.False(t, updated)
+}
+
+func TestController_ReconcileIstio2(t *testing.T) {
```

Comment:
why?
what is the difference between `TestController_ReconcileIstio` and `TestController_ReconcileIstio2` 😂 ?

Reply:
Ups 😅